ee/agent/storage/sqlite/logstore_sqlite.go

```diff
@@ -0,0 +1,103 @@
+package agentsqlite
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+func (s *sqliteStore) getColumns() *sqliteColumns {
+	switch s.tableName {
+	case StartupSettingsStore.String():
+		return &sqliteColumns{pk: "name", valueColumn: "value"}
+	case WatchdogLogStore.String():
+		return &sqliteColumns{pk: "timestamp", valueColumn: "log"}
+	}
+
+	return nil
+}
+
+func (s *sqliteStore) AppendValue(timestamp int64, value []byte) error {
+	colInfo := s.getColumns()
+	if s == nil || s.conn == nil || colInfo == nil {
+		return errors.New("store is nil")
+	}
+
+	if s.readOnly {
+		return errors.New("cannot perform update with RO connection")
+	}
+
+	insertSql := fmt.Sprintf(
+		`INSERT INTO %s (%s, %s) VALUES (?, ?)`,
+		s.tableName,
+		colInfo.pk,
+		colInfo.valueColumn,
+	)
+
+	if _, err := s.conn.Exec(insertSql, timestamp, value); err != nil {
+		return fmt.Errorf("appending row into %s: %w", s.tableName, err)
+	}
+
+	return nil
+}
+
+func (s *sqliteStore) DeleteRows(rowids ...any) error {
+	if s == nil || s.conn == nil {
+		return errors.New("store is nil")
+	}
+
+	if s.readOnly {
+		return errors.New("cannot perform deletes with RO connection")
+	}
+
+	if len(rowids) == 0 {
+		return nil
+	}
+
+	// interpolate the proper number of question marks
+	paramQs := strings.Repeat("?,", len(rowids))
+	paramQs = paramQs[:len(paramQs)-1]
+	deleteSql := fmt.Sprintf(`DELETE FROM %s WHERE rowid IN (%s)`, s.tableName, paramQs)
+
+	if _, err := s.conn.Exec(deleteSql, rowids...); err != nil {
+		return fmt.Errorf("deleting row from %s: %w", s.tableName, err)
+	}
+
+	return nil
+}
+
+func (s *sqliteStore) ForEach(fn func(rowid, timestamp int64, v []byte) error) error {
+	colInfo := s.getColumns()
+	if s == nil || s.conn == nil || colInfo == nil {
+		return errors.New("store is nil")
+	}
```

Comment:
I think you also want a restriction on calling this for a store that isn't the logstore, since scanning data into `var timestamp int64` on line 91 would fail with a different table schema?

Reply:
oh yeah that's a good call! I can probably expand on `getColumns` to either gate usage here or make it work for either, i'll take a look